lib/thunder/pages/thunder_page.dart

```diff
@@ -78,9 +91,83 @@ class _ThunderState extends State<Thunder> {
   @override
   void dispose() {
     pageController.dispose();
+    textIntentDataStreamSubscription.cancel();
+    mediaIntentDataStreamSubscription.cancel();
     super.dispose();
   }
 
+// All listeners to listen Sharing media files & text
+  void handleSharedFilesAndText() {
+    try {
+      handleSharedImages();
+      handleSharedText();
+    } catch (e) {
+      if (context.mounted) showSnackbar(context, AppLocalizations.of(context)!.unexpectedError);
+    }
+  }
+
+  void handleSharedImages() async {
+    // For sharing images from outside the app while the app is closed
+    final initialMedia = await ReceiveSharingIntent.getInitialMedia();
+    if (initialMedia.isNotEmpty) {
+      _navigateToCreatePostPage(image: File(initialMedia.first.path));
+    }
+    // For sharing images while the app is in the memory
+    mediaIntentDataStreamSubscription = ReceiveSharingIntent.getMediaStream().listen((
+      List<SharedMediaFile> value,
+    ) {
+      _navigateToCreatePostPage(image: File(value.first.path));
+    });
+  }
+
+  void handleSharedText() async {
+    // For sharing URLs/text from outside the app while the app is closed
+    final initialText = await ReceiveSharingIntent.getInitialText();
+    if (initialText?.isNotEmpty ?? false) {
+      _navigateToCreatePostPage(text: initialText);
+    }
+
+    // For sharing URLs/text while the app is in the memory
+    textIntentDataStreamSubscription = ReceiveSharingIntent.getTextStream().listen((
+      String? value,
+    ) {
+      if (value?.isNotEmpty ?? false) {
+        _navigateToCreatePostPage(text: value);
+      }
+    });
+  }
+
+  Future<void> _navigateToCreatePostPage({String? text, File? image}) async {
```

Comment:
Could this be moved to a central place (maybe `lib\feed\utils\post.dart`??) as I was thinking I could reuse it for crossposts.

Reply:
Done.